smol/ttree.ml

```diff
@@ -11,6 +11,9 @@ type _ term =
   | TT_forall : { param : typed pat; return : _ term } -> core term
   | TT_lambda : { param : typed pat; return : _ term } -> core term
   | TT_apply : { lambda : _ term; arg : _ term } -> core term
+  | TT_self : { bound : _ pat; body : _ term } -> core term
+  | TT_fix : { bound : _ pat; body : _ term } -> core term
```

Comment:
Self is intentionally not a typed pattern, but fix should be typed, this is a bug that will be fixed in a future PR.